config/database.yml

```diff
@@ -19,28 +19,44 @@ default: &default
   # https://guides.rubyonrails.org/configuring.html#database-pooling
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
 
-development:
-  <<: *default
-  database: eths_indexer_development
+  # Connect on a TCP socket. Omitted by default since the client uses a
+  # domain socket that doesn't need configuration. Windows does not have
+  # domain sockets, so uncomment these lines.
+  host: <%= ENV["DATABASE_HOST"] %>
+
+  # The TCP port the server listens on. Defaults to 5432.
+  # If your server runs on a different port number, change accordingly.
+  port: <%= ENV["DATABASE_PORT"] %>
 
   # The specified database role being used to connect to PostgreSQL.
   # To create additional roles in PostgreSQL see `$ createuser --help`.
   # When left blank, PostgreSQL will use the default role. This is
   # the same name as the operating system user running Rails.
-  #username: eths_indexer
+  username: <%= ENV["DATABASE_USERNAME"] %>
```

Comment:
Will this work in the case the username and password are blank? I think this is the default setup for a local postgres. Likewise I don't think the user should have to deal with a port in most cases. Thinking about this more I'm not sure there is a one-size-fits all simplification for this config file. Maybe we should just add database.yml to gitignore and let people edit it directly.

Thoughts?

Reply:
I've also had some new thoughts. The configuration file should cater to the needs of two types of users:

1.Novices or non-technical individuals who don't need to do much configuration to run the indexer directly.
2.Professional technical personnel who can easily fill in the configurations they need.

Additionally, there's a new consideration:
We can make the ethscriptions-indexer as a product similar to an Ethereum client, where users don't need to worry about internal implementation details. They only need to focus on the RPC interface documentation, and all data requests are made through the RPC interface.

Providing a comprehensive RPC interface and documentation seems like a good choice. This way, users don't need to learn Ruby and PostgreSQL; they can use ethscriptions-indexer within their existing technical framework.

For now, I'll temporarily close this pull request and submit it again when we have a better solution.😃😃

Reply:
What about this? https://github.com/0xFacet/ethscriptions-indexer/pull/46/files

Reply:
> What about this? https://github.com/0xFacet/ethscriptions-indexer/pull/46/files

This is a good idea